model/src/test/kotlin/licenses/ResolvedLicenseInfoTest.kt

```diff
@@ -143,6 +149,177 @@ class ResolvedLicenseInfoTest : WordSpec() {
+
+                val violations = listOf(
+                    RuleViolation(
+                        "example rule",
+                        Identifier("Maven", "org.oss-review-toolkit", "example", "0.0.1"),
+                        SpdxSingleLicenseExpression.parse(gpl),
+                        null,
+                        Severity.ERROR,
+                        "example message",
+                        "how to fix"
+                    )
+                )
+
+                val resolutions = listOf(
+                    RuleViolationResolution(
+                        "example message", RuleViolationResolutionReason.CANT_FIX_EXCEPTION, "example comment"
+                    ),
+                    RuleViolationResolution(
+                        "different regex", RuleViolationResolutionReason.CANT_FIX_EXCEPTION, "example comment"
+                    )
+                )
+
+                val filteredLicenses = resolvedLicenseInfo.licenses.filterLicenseResolutions(
+                    Identifier("Maven", "org.oss-review-toolkit", "example", "0.0.1"),
+                    violations,
+                    resolutions
+                )
+
+                filteredLicenses should beEmpty()
+            }
+
+            "only remove license if it is the licenses of the specified package" {
+                val resolvedLicenses = listOf(
+                    ResolvedLicense(
+                        license = gpl.toSpdx() as SpdxSingleLicenseExpression,
+                        originalDeclaredLicenses = setOf(gpl),
+                        originalExpressions = mapOf(
+                            LicenseSource.DECLARED to setOf(gpl.toSpdx()),
+                            LicenseSource.DETECTED to setOf(gpl.toSpdx())
+                        ),
+                        locations = emptySet()
+                    ),
+                    ResolvedLicense(
+                        license = bsd.toSpdx() as SpdxSingleLicenseExpression,
+                        originalDeclaredLicenses = setOf(bsd),
+                        originalExpressions = mapOf(
+                            LicenseSource.DECLARED to setOf(bsd.toSpdx()),
+                            LicenseSource.DETECTED to setOf(bsd.toSpdx())
+                        ),
+                        locations = emptySet()
+                    ),
+                )
+
+                val violations = listOf(
+                    RuleViolation(
+                        "example rule",
+                        Identifier("Maven", "org.oss-review-toolkit", "example", "0.0.1"),
+                        SpdxSingleLicenseExpression.parse(gpl),
+                        null,
+                        Severity.ERROR,
+                        "example message",
+                        "how to fix"
+                    ),
+                    RuleViolation(
+                        "example rule",
+                        Identifier("Maven", "org.oss-review-toolkit", "example", "0.0.10"),
```

Comment:
It is easy to overlook that there are two different identifiers involved, because the only difference in the coordinates is a digit in the version. Maybe use a completely different identifier for the second package to make this more explicit?